dsp-role/repo/src/test/scala/dsp/role/repo/impl/RoleRepoImplSpec.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.role.repo.impl
+
+import dsp.role.api.RoleRepo
+import dsp.role.domain.RoleTestData
+import dsp.role.repo.impl.RoleRepoLive
+import dsp.role.repo.impl.RoleRepoMock
+import zio._
+import zio.test.Assertion._
+import zio.test._
+
+/**
+ * This spec is used to test all [[RoleRepo]] implementations.
+ */
+object RoleRepoImplSpec extends ZIOSpecDefault {
+
+  def spec = (roleRepoMockTests + roleRepoLiveTests)
+
+  private val role1 = RoleTestData.role1
+  private val role2 = RoleTestData.role2
+
+  val roleTests =
+    test("store two roles and retrieve them all") {
+      for {
+        role1 <- role1.toZIO
+        role2 <- role2.toZIO
+        _     <- RoleRepo.storeRole(role1)
+        _     <- RoleRepo.storeRole(role2)
+        roles <- RoleRepo.getRoles()
+      } yield assertTrue(roles.size == 2)
+    } +
```

Comment:
is there a reason for combining those with plus, which leads to increasing indentation?

Reply:
In this construct the suites are defined on the layer level - see bottom vals. At the end it doesn't bring much, only indentions, and one more level for only one of the layer suites, which is strange. Also in both ways the `+ RoleRepoLive` is not printed out...

```
    + store two roles and retrieve them all
    + store and delete a role
    + store a role and get it by ID
+ RoleRepoMock
  + roleTests
    + store and delete a role
    + store a role and get it by ID
    + store two roles and retrieve them all
 ```